File: Tree/129.Sum-Root-to-Leaf-Numbers/129.Sum-Root-to-Leaf-Numbers.ts
function sumNumbers(root: TreeNode | null): number {
  let ret = 0;
  dfs(root);
  return ret;
  function dfs(root: TreeNode | null, sum: number = 0) {
    if (root === null) {
      return;
    }

    if (root.left === null && root.right === null) {
      ret += sum * 10 + root.val;
    }

    dfs(root.left, sum * 10 + root.val);
    dfs(root.right, sum * 10 + root.val);
  }
}

File: Tree/129.Sum-Root-to-Leaf-Numbers/129.Sum-Root-to-Leaf-Numbers.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def sumNumbers(self, root: Optional[TreeNode]) -> int:
        result = 0
        def dfs(root, sum):
            nonlocal result
            if not root:
                return

            sum = sum * 10 + root.val
            if root.left == None and root.right == None:
                result += sum
            dfs(root.left, sum)
            dfs(root.right, sum)
        dfs(root, 0)
        return result
